### PR TITLE
Removing defaultValue

### PR DESCRIPTION
**Pull request by ntfshard**
Removing deprecated `defaultValues`, technically breaking changes

### CODE REVIEW

**Review by veerajsk — APPROVED** [review]

File: alica_engine/include/engine/blackboard/Blackboard.h [hunk 1]
```diff
@@ -360,34 +347,34 @@ class LockedBlackboardRO
...
-        return _impl->get<T>(key);
+        return _impl.get<T>(key);
     }
-    bool hasValue(const std::string& key) const { return _impl->hasValue(key); }
+    bool hasValue(const std::string& key) const { return _impl.hasValue(key); }
 
 private:
     std::shared_lock<std::shared_mutex> _lk;
-    const BlackboardImpl* _impl;
+    const BlackboardImpl& _impl;
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/blackboard/Blackboard.h:369]: Better to do refactor in separate PR

**Review by Dale-Koenig — APPROVED** [review]

File: alica_engine/include/engine/blackboard/Blackboard.h (hunk 1 above)
Comment (ntfshard) [reply alica_engine/include/engine/blackboard/Blackboard.h:369]: Agree, but it too small
